src/Features/CSharp/Portable/CodeRefactorings/InlineTemporary/InlineTemporaryCodeRefactoringProvider.cs

```diff
@@ -209,6 +212,72 @@ private static async Task<Document> InlineTemporaryAsync(Document document, Vari
+        {
+            var editor = new SyntaxEditor(scope, document.Project.Solution.Services);
+
+            foreach (var identifier in conflictReferences)
+                editor.ReplaceNode(identifier, identifier.WithAdditionalAnnotations(CreateConflictAnnotation()));
+
+            foreach (var identifier in nonConflictReferences)
+            {
+                if (identifier.Parent is AnonymousObjectMemberDeclaratorSyntax { NameEquals: null } anonymousMember)
+                {
+                    // Special case inlining into anonymous types to ensure that we keep property names:
+                    //
+                    // E.g.
+                    //     int x = 42;
+                    //     var a = new { x; };
+                    //
+                    // Should become:
+                    //     var a = new { x = 42; };
+                    editor.ReplaceNode(
+                        anonymousMember,
+                        anonymousMember.Update(NameEquals(identifier), expressionToInline).WithTriviaFrom(anonymousMember));
+                }
+                else if (identifier.Parent is ArgumentSyntax
+                {
+                    Parent: TupleExpressionSyntax tupleExpression,
+                    NameColon: null,
+                } argument &&
+                    !SyntaxFacts.IsReservedTupleElementName(identifier.Identifier.ValueText) &&
+                    tupleExpression.Arguments.Count(a => nonConflictReferences.Contains(a.Expression)) == 1)
+                {
+                    // Special case inlining into tuples to ensure that we keep property names:
+                    //
+                    // E.g.
+                    //     int x = 42;
+                    //     var a = (x, y);
+                    //
+                    // Should become:
+                    //     var a = (x: 42, y);
+                    editor.ReplaceNode(
+                        argument,
+                        argument.Update(NameColon(identifier), argument.RefKindKeyword, expressionToInline).WithTriviaFrom(argument));
+                }
+                else if (identifier.Parent is SpreadElementSyntax spreadElement &&
+                    declarator.Initializer.Value is CollectionExpressionSyntax collectionToInline)
+                {
+                    // Special case inlining a collection into a spread element.  We can just move the original elements
+                    // into the final collection.
+                    var leadingTrivia = spreadElement.GetLeadingTrivia() is [.., (kind: SyntaxKind.WhitespaceTrivia) space]
+                        ? TriviaList(ElasticMarker, space)
+                        : TriviaList(ElasticMarker);
+
+                    editor.ReplaceNode(
+                        spreadElement,
+                        (_, _) => collectionToInline.Elements.Select(
+                            e => e.WithLeadingTrivia(leadingTrivia).WithoutTrailingTrivia()));
```

Comment:
Why do we want to remove trailing trivia here?

Reply:
Probably not needed. Will remove!